os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -98,6 +103,66 @@ int binary_manager_get_index_with_name(char *bin_name)
 	return ERROR;
 }
 
+/****************************************************************************
+ * Name: binary_manager_get_available_size
+ *
+ * Description:
+ *	 This function gets available size in file system.
+ *
+ ****************************************************************************/
+int binary_manager_get_available_size(int bin_idx)
+{
+	DIR *dirp;
+	int size;
+	int name_len;
+	char *bin_name;
+	struct stat file_buf;
+	struct statfs fs_buf;
+	char filepath[CONFIG_PATH_MAX];
+	char running_file[NAME_MAX];
+
+	snprintf(running_file, NAME_MAX, "%s_%s", BIN_NAME(bin_idx), BIN_VER(bin_idx));
+	bin_name = BIN_NAME(bin_idx);
+	name_len = strlen(bin_name);
+	size = 0;
+
+	/* Get available size on file system */
+	if (statfs(BINARY_MNT_PATH, &fs_buf) != OK) {
+		bmdbg("Failed to stat fs %s, errno %d\n", BINARY_MNT_PATH, errno);
+		return ERROR;
+	}
+	size = fs_buf.f_bavail * fs_buf.f_bsize;
```

Comment:
In file system level, this is correct info.
But for user, it is not correct available size because of header of each file system. (please see smart_getllformat in smart.c)
You should use stat additionally

```
struct stat st;
stat("/mnt", &st);
st.st_blksize * fs_buf.f_bsize;